### PR TITLE
fix: defaultQuery getting called with undefined values

### PR DESCRIPTION
**Pull request by ShahAnuj2610**
test with MultiList: https://recordit.co/nSp92Onsuv

### CODE REVIEW

File: src/utils/helper.js [hunk 1]
```diff
@@ -457,15 +455,15 @@ export const updateInternalQuery = (
...
+export const extractQueryFromDefaultQuery = (defaultQuery, props) => {
 	let queryToBeReturned = {};
 	if (defaultQuery) {
 		const evaluateQuery = defaultQuery();
 		if (evaluateQuery) {
 			// we should only retrieve and set the query options here.
 			// [Not implemented yet] `query` key should be handled separately for
 			// adding it to `queryList` in the redux store
-			const { query, ...options } = evaluateQuery;
+			const { query, ...options } = evaluateQuery([], props);
```
Comment (bietkul) [inline src/utils/helper.js]: @ShahAnuj2610 Why the value is an empty array?
Use the `current value` which is specific to the component.

File: src/utils/helper.js [hunk 2]
```diff
@@ -457,15 +455,15 @@ export const updateInternalQuery = (
 	}
 };
 // extracts query options from defaultQuery if set
-export const extractQueryFromDefaultQuery = (defaultQuery) => {
+export const extractQueryFromDefaultQuery = (defaultQuery, props) => {
 	let queryToBeReturned = {};
 	if (defaultQuery) {
 		const evaluateQuery = defaultQuery();
```
Comment (bietkul) [inline src/utils/helper.js]: @ShahAnuj2610 Make the code readable, it's hard to understand how you're passing `defaultQuery`.

**Review by bietkul — CHANGES_REQUESTED** [review]

**Comment by bietkul** [comment]
@ShahAnuj2610 Any update here?

**Comment by ShahAnuj2610** [comment]
> @ShahAnuj2610 Any update here?

@bietkul fixing it today and updating the related PR.

**Comment by ShahAnuj2610** [comment]
@bietkul have attached test video for this and it's related PR

**Review by bietkul — APPROVED** [review]

**Pull request merged**